src/components/tooltip/tooltip.component.tsx

```diff
@@ -73,6 +91,127 @@ export const Tooltip = React.forwardRef<HTMLDivElement | null, TooltipProps>(
     }: TooltipProps,
     ref
   ) => {
+    const isControlled = isVisible !== undefined;
+
+    const [isOpen, setIsOpen] = useState(false);
+
+    let showTooltip = isOpen;
+    if (isControlled) {
+      showTooltip = isVisible;
+    }
+
+    const flipOverridesRef = useRef(flipOverrides);
+    flipOverridesRef.current = flipOverrides;
+
+    const arrowReference = useRef<HTMLDivElement | null>(null);
+
+    const calculateOffset = useCallback(
+      (placement) => {
+        let mainAxis = 9;
+
+        if (isPartOfInput) {
+          const inputSizes = {
```

Comment:
Would it be worth making this a function and moving it out of the scope of the component so it's not instantiated each time a tooltip is rendered? It's very trivial so I'm happy to leave as is if you disagree


```
mainAxis = inputSizes(placement, inputSize);
```

Reply:
I don't think it's worth it to be honest, sounds like a micro optimization